justfile

```diff
@@ -31,39 +30,75 @@ docs:
 
 # Default build
 build:
-    cargo build --workspace --all-targets
+    cargo build --all-targets --workspace
 
 # Run all tests
-test: build
-    cargo test --workspace --all-targets
+test *ARGS: build
+    cargo test --all-targets --workspace {{ARGS}}
 
-# Test documentation
-test-doc:
-    cargo doc --no-deps
+# Find unused dependencies. Install it with `cargo install cargo-udeps`
+udeps:
+    cargo +nightly udeps --all-targets --workspace
 
 rust-info:
     rustc --version
     cargo --version
 
 # Run all tests as expected by CI
-ci-test: rust-info test-fmt clippy test test-doc
+ci-test: rust-info test-fmt clippy test
 
 # Verify that the current version of the crate is not the same as the one published on crates.io
-check-if-published:
+check-if-published CRATE_NAME="vmod-rers":
     #!/usr/bin/env bash
-    LOCAL_VERSION="$(grep '^version =' Cargo.toml | sed -E 's/version = "([^"]*)".*/\1/')"
-    echo "Detected crate version:  $LOCAL_VERSION"
-    CRATE_NAME="$(grep '^name =' Cargo.toml | head -1 | sed -E 's/name = "(.*)"/\1/')"
-    echo "Detected crate name:     $CRATE_NAME"
-    PUBLISHED_VERSION="$(cargo search ${CRATE_NAME} | grep "^${CRATE_NAME} =" | sed -E 's/.* = "(.*)".*/\1/')"
+    set -euo pipefail
+    LOCAL_VERSION="$(cargo metadata --format-version 1 | jq -r --arg CRATE_NAME {{quote(CRATE_NAME)}}  '.packages | map(select(.name == $CRATE_NAME)) | first | .version')"
+    echo "Detected crate {{CRATE_NAME}} version:  $LOCAL_VERSION"
+    PUBLISHED_VERSION="$(cargo search --quiet {{quote(CRATE_NAME)}} | grep "^{{CRATE_NAME}} =" | sed -E 's/.* = "(.*)".*/\1/')"
     echo "Published crate version: $PUBLISHED_VERSION"
     if [ "$LOCAL_VERSION" = "$PUBLISHED_VERSION" ]; then
         echo "ERROR: The current crate version has already been published."
         exit 1
     else
         echo "The current crate version has not yet been published."
     fi
+
+[private]
+docker-build-ver VERSION:
+    docker build \
+           --progress=plain \
+           -t "varnish-img-{{VERSION}}" \
+           {{ if VERSION == "latest" { "" } else { "--build-arg VARNISH_VERSION_TAG=varnish" + VERSION } }} \
+           --build-arg USER_UID=$(id -u) \
+           --build-arg USER_GID=$(id -g) \
+           -f docker/Dockerfile \
+           .
+
+[private]
+docker-run-ver VERSION *ARGS:
+    mkdir -p docker/.cache/{{VERSION}}
+    touch docker/.cache/{{VERSION}}/.bash_history
+    docker run --rm -it \
+        -v "$PWD:/app/" \
+        -v "$PWD/docker/.cache/{{VERSION}}:/home/user/.cache" \
+        -v "$PWD/docker/.cache/{{VERSION}}/.bash_history:/home/user/.bash_history" \
+        varnish-img-{{VERSION}} {{ARGS}}
+
+docker-run-latest *ARGS: (docker-build-ver "latest") (docker-run-ver "latest" ARGS)
+docker-run-77 *ARGS: (docker-build-ver "77") (docker-run-ver "77" ARGS)
+docker-run-76 *ARGS: (docker-build-ver "76") (docker-run-ver "76" ARGS)
+docker-run-60 *ARGS: (docker-build-ver "60lts") (docker-run-ver "60lts" ARGS)
```

Comment:
we don't support 60lts (requires VDP integration that's disabled for this version)